lib/elixir/lib/kernel/lexical_tracker.ex

```diff
@@ -98,61 +116,79 @@ defmodule Kernel.LexicalTracker do
 
   # Callbacks
 
-  def init(dest) do
-    {:ok, {:ets.new(__MODULE__, [:protected]), dest}}
+  def init({dest, parent}) do
+    {:ok, {:ets.new(__MODULE__, [:protected]), dest, parent}}
   end
 
   @doc false
-  def handle_call(:ets, _from, {d, dest}) do
-    {:reply, d, {d, dest}}
+  def handle_call(:ets, _from, {d, dest, parent}) do
+    {:reply, d, {d, dest, parent}}
   end
 
-  def handle_call(:dest, _from, {d, dest}) do
-    {:reply, dest, {d, dest}}
+  def handle_call(:dest, _from, {d, dest, parent}) do
+    {:reply, dest, {d, dest, parent}}
   end
 
-  def handle_cast({:remote_dispatch, module, mode}, {d, dest}) do
+  def handle_cast({:remote_dispatch, module, mode}, {d, dest, parent}) do
+    if parent, do: remote_dispatch(parent, module, mode)
+
     add_reference(d, module, mode)
-    {:noreply, {d, dest}}
+    {:noreply, {d, dest, parent}}
   end
 
-  def handle_cast({:import_dispatch, {module, function, arity}}, {d, dest}) do
+  def handle_cast({:remote_dispatch, module, fas, line, mode}, {d, dest, parent}) do
+    if parent, do: remote_dispatch(parent, module, fas, line, mode)
+
+    add_reference(d, module, mode)
+    add_reference(d, module, fas, line, mode)
+    {:noreply, {d, dest, parent}}
+  end
+
+  def handle_cast({:import_dispatch, {module, function, arity}}, {d, dest, parent}) do
```

Comment:
Maybe could add it as another element of `remotes/1` that simply lists the functions that were imported and where they were dispatched from? Still not sure what it would be used for, though.